sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -170,6 +170,25 @@ case class CatalogTablePartition(
   }
 }
 
+/**
+ * A container for clustering information.
+ *
+ * @param columnNames the names of the columns used for clustering.
+ */
+case class ClusterBySpec(columnNames: Seq[UnresolvedAttribute]) {
```

Comment:
It's weird to use `UnresolvedAttribute` here as we are not going to resolve it in the analyzer. How about just `Seq[Seq[String]]` or `Seq[FieldReference]`?